ui/src/shared/utils/convertUserInput.test.ts

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import {convertUserInputToNumOrNaN} from 'src/shared/utils/convertUserInput'
+
+describe('convertUserInputToNumOrNaN', () => {
+  let value
+  let result
+  const event = {
+    target: {
+      value: '',
+    },
+  } as React.ChangeEvent<HTMLInputElement>
+
+  it('should convert a number string to a number', () => {
+    value = 123
+    event.target.value = `${value}`
+    expect(convertUserInputToNumOrNaN(event)).toEqual(value)
+  })
+
+  it('should convert a non-numeric, non-empty string to NaN', () => {
+    value = 'a number'
+    event.target.value = value
+
+    result = convertUserInputToNumOrNaN(event)
+    expect(Number.isNaN(result)).toEqual(true)
+  })
+
+  it('should convert an empty string to NaN', () => {
+    value = ''
+    event.target.value = value
+
+    result = convertUserInputToNumOrNaN(event)
+    expect(Number.isNaN(result)).toEqual(true)
+  })
+})
```

Comment:
nice tests!

what do you think about adding tests for some falsey values like `null`, `undefined`, and `false` since it seems like from below, the code handles those cases?

Reply:
Yes, will do.